drivers/sensor/lps28dfw/lps28dfw.c

```diff
@@ -0,0 +1,332 @@
+/* ST Microelectronics LPS28DFW pressure and temperature sensor
+ *
+ * Copyright (c) 2023 STMicroelectronics
+ * Copyright (c) 2023 PHYTEC Messtechnik GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/lps28dfw.pdf
+ */
+
+#define DT_DRV_COMPAT st_lps28dfw
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/logging/log.h>
+
+#include "lps28dfw.h"
+
+#define LPS28DFW_SWRESET_WAIT_TIME	50
+
+LOG_MODULE_REGISTER(LPS28DFW, CONFIG_SENSOR_LOG_LEVEL);
+
+static inline int lps28dfw_set_odr_raw(const struct device *dev, uint8_t odr)
+{
+	const struct lps28dfw_config * const cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lps28dfw_md_t md;
+
+	md.odr = odr;
+	md.avg = cfg->avg;
+	md.lpf = cfg->lpf;
+	md.fs = cfg->fs;
+
+	return lps28dfw_mode_set(ctx, &md);
+}
+
+static int lps28dfw_sample_fetch(const struct device *dev,
+				enum sensor_channel chan)
+{
+	struct lps28dfw_data *data = dev->data;
+	const struct lps28dfw_config * const cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lps28dfw_data_t raw_data;
+	lps28dfw_md_t md;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	if (lps28dfw_data_get(ctx, &md, &raw_data) < 0) {
+		LOG_DBG("Failed to read sample");
+		return -EIO;
+	}
+
+	data->sample_press = raw_data.pressure.raw;
+	data->sample_temp = raw_data.heat.raw;
+
+	return 0;
+}
+
+static inline void lps28dfw_press_convert(const struct device *dev,
+					 struct sensor_value *val,
+					 int32_t raw_val)
+{
+	int32_t press_tmp = raw_val >> 8; /* raw value is left aligned (24 msb) */
+	lps28dfw_md_t md;
+
+	md.fs = ((struct lps28dfw_config *)dev->config)->fs;
```

Comment:
how about adding an intermediate variable for `dev->config`? bit easier to read, you already do it in `lps28dfw_init_chip`